File: Lab1/Lab1 Java/src/NumberOperations.java
import java.math.BigDecimal;
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class NumberOperations {

    public static void main(String[] args) {
        ArrayList<Number> numbers = new ArrayList<Number>(Arrays.asList(10, 20.5, 30, 40.7, 50, 60.3, 70, 80.1, 90, 100.9));

        // Завдання
        // 1. Створіть числа різних типів
        byte num0 = 1;
        int num1 = 29;
        short num2 = 30;
        long num3 = 40L;
        float num4 = 3.1415f;
        double num5 = 14.54321;

        // 2. Долучіть їх до списку +
        // 3. Самостійно вирішіть, якого типу має бути список, щоб він містив всі різнотипні числа
        List<Number> newNumbers = Arrays.asList(num0, num1, num2, num3, num4, num5);
        numbers.addAll(newNumbers);

        // 4. Виведіть числа на екран
        System.out.print("Всі числа: ");
        System.out.println(numbers);
        System.out.println();

        // 5. Виведіть числа у форматі цілих чисел
        ArrayList<Number> intNumbers = new ArrayList<>();
        for (Number number : numbers) {
            intNumbers.add(Math.round(number.doubleValue()));
        }
        System.out.print("Округлені числа: ");
        System.out.println(intNumbers);
        System.out.println();

        // 6. Виведіть числа у форматі дробних чисел з 2ма знаками після коми (точки)
        System.out.print("Числа з 2ма знаками після коми: ");
        for (Number number : numbers) {
            System.out.print(String.format("%.2f", number.doubleValue()) + " ");
        }
        System.out.println();
        System.out.println();

        // 7. Придумайте та реалізуйте рішення, щоб зберігати числа в різних списках в
        // залежності від їх типу
        List<Integer> intList = new ArrayList<Integer>();
        List<Byte> byteList = new ArrayList<Byte>();
        List<Short> shortList = new ArrayList<Short>();
        List<Long> longList = new ArrayList<Long>();
        List<Float> floatList = new ArrayList<Float>();
        List<Double> doubleList = new ArrayList<Double>();

        for (Number number : numbers) {
            switch (number) {
                case Integer i -> intList.add((int)number);
                case Byte b -> byteList.add((byte)number);
                case Short i -> shortList.add((short)number);
                case Long l -> longList.add((long)number);
                case Float v -> floatList.add((float)number);
                case Double v -> doubleList.add((double)number);
                case null, default -> {
                    continue;
                }
            }
        }

        System.out.println(
            "Масив Int: " + intList + "\n" +
            "Масив Byte: " + byteList + "\n" +
            "Масив Short: " + shortList + "\n" +
            "Масив Long: " + longList + "\n" +
            "Масив Float: " + floatList + "\n" +
            "Масив Double: " + doubleList
        );
        System.out.println();

        // 9й варіант. Створіть новий список, де кожне число початкового списку помножене на 2
        numbers = new ArrayList<Number>(
                Arrays.asList(10, 20.5, 30, 40.7, 50, 60.3, 70, 80.1, 90, 100.9)
        );
        ArrayList<Number> doubledNumbers = new ArrayList<>();
        for (Number number : numbers) {
            if (number instanceof Double) {
                doubledNumbers.add(number.doubleValue() * 2);
            } else {
                doubledNumbers.add(number.intValue() * 2);
            }
        }
        System.out.print("Числа помножені на 2: ");
        System.out.println(doubledNumbers);
        for (Number number : doubledNumbers) {
            System.out.print(String.format("%.2f", number.doubleValue()) + " ");
        }

        BigInteger number1 = new BigInteger("100000000000000");
        BigDecimal number2 = new BigDecimal("100000.123456789");

        BigDecimal number3 = new BigDecimal(number1);
        System.out.println(number2.add(number3));
    }
}

File: Lab1/Lab1 C#/NumberOperations.cs
using System;
using System.Collections.Generic;
//using Extreme.Mathematics;
using System.Numerics;
using Extreme.Mathematics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab1_C_
{
    internal class NumberOperations
    {
        static void Main(string[] args)
        {
            List<dynamic> numbers = new List<dynamic>() { 10, 20.5, 30, 40.7, 50, 60.3, 70, 80.1, 90, 100.9 };

            // Завдання
            // 1. Створіть числа різних типів
            byte num0 = 1;
            int num1 = 29;
            short num2 = 30;
            long num3 = 40L;
            float num4 = 3.1415f;
            double num5 = 14.54321;

            // 2. Долучіть їх до списку + 3. Самостійно вирішіть, якого типу має бути список, щоб він містив всі різнотипні числа
            List<dynamic> newNumbers = new List<dynamic>() { num0, num1, num2, num3, num4, num5 };
            newNumbers.ForEach(n => numbers.Add(n));

            // 4. Виведіть числа на екран
            Console.WriteLine($"Всi числа: {String.Join("; ", numbers.ToArray())}\n");

            // 5. Виведіть числа у форматі цілих чисел
            List<dynamic> intNumbers = new List<dynamic>();
            foreach (dynamic number in numbers)
            {
                if (number is double || number is float) intNumbers.Add(Math.Round((double)number));
                else intNumbers.Add(number);
            }
            Console.WriteLine($"Округленi числа: {String.Join("; ", intNumbers.ToArray())}\n");

            // 6. Виведіть числа у форматі дробних чисел з 2ма знаками після коми (точки)
            Console.Write("Числа з 2ма знаками пiсля коми: ");
            foreach (dynamic number in numbers) Console.Write(String.Format("{0:0.00}", number) + "; ");
            Console.WriteLine("\n");

            // 7. Придумайте та реалізуйте рішення, щоб зберігати числа в різних списках в залежності від їх типу
            List<int> intList = new List<int>();
            List<byte> byteList = new List<byte>();
            List<short> shortList = new List<short>();
            List<long> longList = new List<long>();
            List<float> floatList = new List<float>();
            List<double> doubleList = new List<double>();

            foreach (dynamic number in numbers)
            {
                if (number is int) intList.Add(number);
                else if (number is byte) byteList.Add(number);
                else if (number is short) shortList.Add(number);
                else if (number is long) longList.Add(number);
                else if (number is float) floatList.Add(number);
                else if (number is double) doubleList.Add(number);
                else continue;
            }

            Console.WriteLine(
                $"Масив Int: {String.Join("; ", intList.ToArray())} \n" +
                $"Масив Byte: {String.Join("; ", byteList.ToArray())} \n" +
                $"Масив Short: {String.Join("; ", shortList.ToArray())} \n" +
                $"Масив Long: {String.Join("; ", longList.ToArray())} \n" +
                $"Масив Float: {String.Join("; ", floatList.ToArray())} \n" +
                $"Масив Double: {String.Join("; ", doubleList.ToArray())} \n"
            );

            // 9й варіант. Створіть новий список, де кожне число початкового списку помножене на 2
            numbers = new List<dynamic>() { 10, 20.5, 30, 40.7, 50, 60.3, 70, 80.1, 90, 100.9 };
            List<dynamic> doubledNumbers = new List<dynamic>();
            foreach (dynamic number in numbers) doubledNumbers.Add(number * 2);
            Console.WriteLine($"Числа помноженi на 2: {String.Join("; ", doubledNumbers.ToArray())}");

            System.Numerics.BigInteger bigNumber = System.Numerics.BigInteger.Parse("100000000000000000000");
            ExtendedNumerics.BigDecimal bigDouble = ExtendedNumerics.BigDecimal.Parse("12345678901234567890,123456789");

            Console.WriteLine(bigNumber + bigDouble);
        }
    }
}
